autobot/telegram/parser.py

```diff
@@ -238,16 +238,139 @@ def _entity_parser (entity:dict):
+            game_obj = Game()
+            for k,v in val.items():
+                match k:
+                    case "photo":
+                        pht = self._parse_photosize(k,v)
+                        setattr(game_obj,k,pht)
+                    case "animation":
+                        anim = self._parse_animation(k,v)
+                        setattr(game_obj,k,anim)
+                    case _:
+                        setattr(game_obj,k,v)
+
+        elif (key == "game_high_score"):
+            game_obj = GameHighScore()
+            for k,v in val.items():
+                setattr(game_obj,k,v)
+
+        return game_obj
     
     
     # methods to parse inline objects
+    def _parse_inline (self,key:str,val:dict) -> InlineKeyboardMarkup|InlineQueryResult|InputMessageContent:
+        inline_obj = None
+
+        if (key == "reply_markup"):
+            inline_obj = InlineKeyboardMarkup()
+            for k,v in val.items():
+                match k:
+                    case "inline_keyboard":
+                        inline_keyboard = self._parse_inlinekeyboardbutton(k,v)
+                        setattr(inline_obj,k,inline_keyboard)
+                    case _:
+                        setattr(inline_obj,k,v)
+
+        elif (key == "inline_query_result"):
+            inline_obj = InlineQueryResult()
+            for k,v in val.items():
+                match k:
+                    case "thumb_url"|"thumb_width"|"thumb_height":
+                        thumb = self._parse_thumb(k,v)
+                        setattr(inline_obj,k,thumb)
+                    case "input_message_content":
+                        input_msg = self._parse_inputmessagecontent(k,v)
+                        setattr(inline_obj,k,input_msg)
+                    case "reply_markup":
+                        inline_keyboard = self._parse_inlinekeyboardmarkup(k,v)
+                        setattr(inline_obj,k,inline_keyboard)
+                    case _:
+                        setattr(inline_obj,k,v)
+
+        elif (key == "input_message_content"):
+            inline_obj = InputMessageContent()
+            for k,v in val.items():
+                match k:
+                    case "location":
+                        loc = self._parse_location(k,v)
+                        setattr(inline_obj,k,loc)
+                    case "venue":
+                        ven = self._parse_venue(k,v)
+                        setattr(inline_obj,k,ven)
+                    case "contact":
+                        cont = self._parse_contact(k,v)
+                        setattr(inline_obj,k,cont)
+                    case _:
+                        setattr(inline_obj,k,v)
+
+        return inline_obj
+    
     
     
     # methods to parse passport objects
+    def _parse_passport (self,key:str,val:dict) -> PassportData:
+        passport_obj = PassportData()
+        for k,v in val.items():
+            match k:
+                case "encrypted_data":
```

Comment:
I see as if PassportData has only "credentials" and "data" properties. I can't find "encrypted_data".